peps/pep-0788.rst

```diff
@@ -631,6 +642,54 @@ in the C API documentation, ideally under the :ref:`python:gilstate` section.
 The existing ``PyGILState`` documentation should be updated accordingly to point
 to the new APIs.
 
+.. _pep-788-porting-guide:
+
+Porting Guide
+-------------
+
+Ensuring Python Can Be Called
+*****************************
+
+In all places where :c:func:`PyGILState_Ensure` and :c:func:`PyGILState_Release`
+are used, provide access to a strong or weak interpreter reference in the code
+path. If this is too difficult or impossible (such as in a public library interface
+that uses ``PyGILState`` under the hood), consider using :c:func:`PyInterpreterRef_Main`
+to get a reference for the main interpreter. Then, replace the legacy calls with
+:c:func:`PyThreadState_Ensure` and :c:func:`PyThreadState_Release`, acquiring
+and releasing interpreter references where necessary. A strong interpreter
+reference should typically be released after calling :c:func:`PyThreadState_Release`.
+
+Thread States
+*************
+
+There are two ``PyGILState`` APIs that act as an abstraction for
+:term:`thread states <thread state>`. For some background, Python stores
```

Comment:
What I mean by “porting guide” is *concrete steps* to do when the compiler gives you deprecation warnings. No background and theory, and as few assumptions as possible about the reader knowing what the code in question does (i.e. if there's a purely mechanical replacement, write that; otherwise list alternatives and say how to choose between them; and say what to watch out for.)
Here's one I wrote recently: https://peps.python.org/pep-0793/#porting-guide

(A porting guide doesn't *need* to be in the PEP, but it should be in the final docs. Might be best to remove it from *this* PR.)